### DISCUSSION TITLE
Any way of preventing zoom (in and out) in browser during experiment?

### DISCUSSION THREAD

**Discussion by yuvalharr** [jspsych/jsPsych#845] [Q&A] [answered]
Hey there,

As in my experiment it is very important that stimuli will be presented in the same size for all subjects, I was wondering whether there is any way to disable browser zooming during the experiment. I'm using the 'resize' plugin and 'fullscreen' plugin, but as long  that zoom is enabled I can't be really sure about the stimuli size.

Thanks!

**Comment by jodeleeuw** [comment, marked as answer]
Hi @yuvalharr,

Not that I know of. I suspect it is not possible because that would get in the way of accessibility, which browsers are interested in maintaining. 

You can detect zoom events though:

```js
var zoom_event = [];

window.onresize = function(){
  console.log(window.devicePixelRatio);
  zoom_event.push({timestamp: performance.now(), zoom_ratio: window.devicePixelRatio});
}

// add this trial to your timeline near the end of the experiment
// this will store the zoom_event data in the data for this trial.
var record_zoom_events = {
  type: 'call-function',
  func: function(){
    return JSON.stringify(zoom_event);
  }
}
```

I'd suggest:

1) Asking participants to not use the zoom feature at the start of the experiment.
2) Recording all zoom events using the method I sketched above. 
3) Exclude participants who use the zoom feature from the analysis.

**Comment by becky-gilbert** [comment]
**Edit:** My suggestion below to add the viewport settings in a `meta` tag only applies on mobile devices, and won't work on desktop/laptop browsers. So @yuvalharr should stick with @jodeleeuw's response. But I'll leave my original response here in case anyone wants to know how to do this on mobile.

----

I think this is possible using a `meta` tag in the html `<head>` section to control the viewport, with `user-scalable` set to `no` or `0`. Something like this:

```html
<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
```

But from [this page](https://sitebulb.com/hints/mobile-friendly/the-viewport-meta-tag-prevents-the-user-from-scaling/) I see that some browsers might choose to ignore the `user-scalable` setting because, as @jodeleeuw says, it interferes with accessibility. So if you use this then I would definitely test it on all the browsers that you expect people to use.

**Reply by jodeleeuw** [reply]
I'm not sure, but this may only work on mobile (and only some mobile browsers may honor it). When I tried using this tag on a Chrome desktop I was still able to zoom in and out.

**Reply by becky-gilbert** [reply]
Ah, yes, you're absolutely right! Apparently the meta viewport settings only apply to mobile devices and are ignored by desktop browsers. I'll edit my response.
